spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -30,8 +30,11 @@ protected String metadataTableName(MetadataTableType type) {
     String tableName = table().toString();
     if (tableName.contains("/")) {
       return tableName + "#" + type;
-    } else if (tableName.startsWith("hadoop.") || tableName.startsWith("hive.")) {
-      // HiveCatalog and HadoopCatalog prepend a logical name which we need to drop for Spark 2.4
+    } else if (tableName.startsWith("hadoop.")) {
+      // Load a path by HadoopCatalog or HadoopTables
+      return table().location() + "#" + type;
+    } else if (tableName.startsWith("hive.")) {
+      // HiveCatalog prepend a logical name which we need to drop for Spark 2.4
       return tableName.replaceFirst("(hadoop\\.)|(hive\\.)", "") + "." + type;
```

Comment:
Looks like this needs to be updated. There is no need to remove `hadoop.` if Hadoop tables don't use this code path.